### PR TITLE
feat(chainHead_v1): operationWaitingForContinue and chainHead_v1_continue

### LINKED ISSUE #837: PAPI storage getEntries() returns only 100 entries
Hi,

First of all thank you for providing us with this great tool! 

While trying to run my polkadot-api (PAPI) based dapp on chopsticks I noticed that on Asset Hub, `api.query.Assets.Metadata.getEntries()` returns only the first 100 items, instead of 300+

I noticed that it can be fixed by wrapping the provider with `withPolkadotSdkCompat`, but I've been told that it shouldn't be necessary.

Repo to reproduce : https://github.com/0xKheops/papi-chopsticks-getentries

I get this output:
image

### PR DESCRIPTION
**Pull request by voliva**
I've added pagination by performing storage queries with 1000 values at a time, then storing the pending requests on the "callbacks" map we had, which I have repurposed to better represent the state of each follow subscription.

I also added a test by querying a storage query that has more than 1000 values, but without being too much. I've found System.BlockHash has 1201 entries, so I thought that's a good candidate. Let me know if I should change this for a better alternative.

Closes #837

### CODE REVIEW

File: packages/core/src/rpc/rpc-spec/chainHead_v1.ts [hunk 1]
```diff
@@ -5,7 +5,17 @@ import { defaultLogger } from '../../logger.js'
 
 const logger = defaultLogger.child({ name: 'rpc-chainHead_v1' })
 
-const callbacks = new Map<string, (data: any) => void>()
+type DescendantValuesParams = {
+  prefix: string
+  startKey: string
+}
+const following = new Map<
```
Comment (xlc) [inline packages/core/src/rpc/rpc-spec/chainHead_v1.ts:12]: One downside about global variables is that for some use cases, we need to handle chopsticks instances running at a same time (e.g. unit tests) so need to ensure they don't interfere with each other. 
In this particular case, it is fine as everything is indexed by a unique random ID. The only drawback I can see is potential memory leak (i.e. destroy a chopsticks instance will not clear entry here). But that's not a very big deal here.
So this is ok for now. Maybe later refactor this into context somehow.

File: packages/core/src/rpc/rpc-spec/chainHead_v1.ts [hunk 2]
```diff
@@ -166,6 +176,48 @@ export interface StorageItemRequest {
...
+  block: Block,
+  params: DescendantValuesParams,
+): Promise<{
+  items: Array<{
+    key: string
+    value?: HexString
+  }>
+  next: DescendantValuesParams | null
+}> {
+  const pageSize = 100
```
Comment (xlc) [inline packages/core/src/rpc/rpc-spec/chainHead_v1.ts:190]: you have 2 page_size and one is 1000 and one is 100?

File: packages/core/src/rpc/rpc-spec/chainHead_v1.ts (hunk 1 above)
Comment (xlc) [reply packages/core/src/rpc/rpc-spec/chainHead_v1.ts:12]: #839

File: packages/core/src/rpc/rpc-spec/chainHead_v1.ts (hunk 2 above)
Comment (voliva) [reply packages/core/src/rpc/rpc-spec/chainHead_v1.ts]: my bad - leftover from previous code.

Thanks for pointing it out!

**Pull request merged**